src/tools/rustdoc-gui/tester.js

```diff
@@ -101,25 +115,88 @@ async function main(argv) {
     let failed = false;
     let files;
     if (opts["files"].length === 0) {
-        files = fs.readdirSync(opts["tests_folder"]).filter(file => path.extname(file) == ".goml");
+        files = fs.readdirSync(opts["tests_folder"]);
     } else {
-        files = opts["files"].filter(file => path.extname(file) == ".goml");
+        files = opts["files"];
+    }
+    files = files.filter(file => path.extname(file) == ".goml");
+    if (files.length === 0) {
+        console.log("rustdoc-gui: No test selected");
+        process.exit(2);
     }
-
     files.sort();
-    for (var i = 0; i < files.length; ++i) {
+
+    console.log(`running ${files.length} rustdoc-gui tests`);
+    process.setMaxListeners(files.length + 1);
+    let tests = [];
+    let results = new Array(files.length);
+    // poormans enum
+    const RUN_SUCCESS = 42, RUN_FAILED = 23, RUN_ERRORED = 13;
+    for (let i = 0; i < files.length; ++i) {
         const testPath = path.join(opts["tests_folder"], files[i]);
-        await runTest(testPath, options).then(out => {
-            const [output, nb_failures] = out;
-            console.log(output);
-            if (nb_failures > 0) {
+        tests.push(
+            runTest(testPath, options)
+            .then(out => {
+                //console.log(i);
+                const [output, nb_failures] = out;
+                results[i] = {
+                    status: nb_failures === 0 ? RUN_SUCCESS : RUN_FAILED,
+                    output: output,
+                };
+                if (nb_failures > 0) {
+                    print_test_erroneous()
+                    failed = true;
+                } else {
+                    print_test_successful()
+                }
+            })
+            .catch(err => {
+                results[i] = {
+                    status: RUN_ERRORED,
+                    output: err,
+                };
+                print_test_erroneous();
                 failed = true;
-            }
-        }).catch(err => {
-            console.error(err);
-            failed = true;
-        });
+            })
+        );
+        if (no_headless) {
+            await tests[i];
```

Comment:
Nice! :+1:

Reply:
My only concern here is that now we don't know which test is running anymore though...

Reply:
I just ran the test with this command:
```bash
./x.py test src/test/rustdoc-gui --stage 1 --jobs 8 --test-args --no-headless
```
For me it is very hard to tell what is going on because the windows pop up, do something for a second or two and then flash away.
Should the dev confirm each run?

Reply:
Sorry, each test

Reply:
Actually it doesn't matter. I guess we're only interesting in the one failing.